src/OrchardCore.Modules/OrchardCore.Media.AmazonS3/AwsStorageOptionsExtension.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using Amazon;
+using Amazon.Runtime.CredentialManagement;
+using Microsoft.Extensions.Configuration;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.FileStorage.AmazonS3;
+
+namespace OrchardCore.Media.AmazonS3;
+
+public static class AwsStorageOptionsExtension
+{
+    public static IEnumerable<ValidationResult> Validate(this AwsStorageOptions options)
+    {
+        if (String.IsNullOrWhiteSpace(options.BucketName))
+        {
+            yield return new ValidationResult(Constants.ValidationMessages.BucketNameIsEmpty);
+        }
+
+        if (options.Credentials != null)
+        {
+            if (String.IsNullOrWhiteSpace(options.Credentials.SecretKey))
+            {
+                yield return new ValidationResult(Constants.ValidationMessages.SecretKeyIsEmpty);
+            }
+
+            if (String.IsNullOrWhiteSpace(options.Credentials.AccessKeyId))
+            {
+                yield return new ValidationResult(Constants.ValidationMessages.AccessKeyIdIsEmpty);
+            }
+
+            if (String.IsNullOrWhiteSpace(options.Credentials.RegionEndpoint))
+            {
+                yield return new ValidationResult(Constants.ValidationMessages.RegionEndpointIsEmpty);
+            }
+
+        }
+    }
+
+    public static AwsStorageOptions BindConfiguration(this AwsStorageOptions options, IShellConfiguration shellConfiguration)
+    {
+        var section = shellConfiguration.GetSection("OrchardCore_Media_Amazon_S3");
```

Comment:
In fact before we would have used `OrchardCore.Media.AmazonS3`.

But for compatibility under linux we needed to replace all `.` by `_`.

So maybe `OrchardCore_Media_AmazonS3`, only to be consistent with the module/feature name.

Reply:
Yeap, changed.